src/main/java/org/opentripplanner/standalone/config/routerequest/RouteRequestConfig.java

```diff
@@ -335,176 +336,52 @@ The board time is added to the time when going from the stop (offboard) to onboa
-          .withUnpreferredVehicleParkingTagCost(
-            c
-              .of("unpreferredVehicleParkingTagCost")
-              .since(V2_3)
-              .summary("What cost to add if a parking facility doesn't contain a preferred tag.")
-              .description("See `preferredVehicleParkingTags`.")
-              .asInt(
-                VehicleParkingPreferences.DEFAULT.unpreferredVehicleParkingTagCost().toSeconds()
-              )
-          )
-          .withBannedVehicleParkingTags(
-            c
-              .of("bannedVehicleParkingTags")
-              .since(V2_1)
-              .summary(
-                "Tags with which a vehicle parking will not be used. If empty, no tags are banned."
-              )
-              .description(
-                """
-                Vehicle parking tags can originate from different places depending on the origin of the parking(OSM or RT feed).
-                """
-              )
-              .asStringSet(List.of())
-          )
-          .withRequiredVehicleParkingTags(
-            c
-              .of("requiredVehicleParkingTags")
-              .since(V2_1)
-              .summary(
-                "Tags without which a vehicle parking will not be used. If empty, no tags are required."
-              )
-              .description(
-                """
-                Vehicle parking tags can originate from different places depending on the origin of the parking(OSM or RT feed).
-                """
-              )
-              .asStringSet(List.of())
-          )
-          .withParkTime(
-            c
-              .of("bikeParkTime")
-              .since(V2_0)
-              .summary("Time to park a bike.")
-              .asDuration(VehicleParkingPreferences.DEFAULT.parkTime())
-          )
-          .withParkCost(
-            c
-              .of("bikeParkCost")
-              .since(V2_0)
-              .summary("Cost to park a bike.")
-              .asInt(VehicleParkingPreferences.DEFAULT.parkCost().toSeconds())
-          )
-          .withPreferredVehicleParkingTags(
-            c
-              .of("preferredVehicleParkingTags")
-              .since(V2_3)
-              .summary(
-                "Vehicle parking facilities that don't have one of these tags will receive an extra cost and will therefore be penalised."
-              )
-              .description(
-                """
-                Vehicle parking tags can originate from different places depending on the origin of the parking(OSM or RT feed).
-                """
-              )
-              .asStringSet(List.of())
+          .description(
+            "If the triangle optimization is used, it's enough to just define the triangle parameters"
           )
+          .asEnum(dft.optimizeType())
       )
-      .withRental(it -> setVehicleRental(c, it));
+      // triangle overrides the optimization type if defined
+      .withForcedOptimizeTriangle(it -> mapOptimizationTriangle(cb, it))
```

Comment:
I'm in favour of this behaviour but doesn't this change how the API works?

Reply:
This is only used by the configuration which we can break now. Later on I will use it for the new plan query as well. All the other places use the old method.